src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1151,14 +1151,30 @@ private IBinaryOperation CreateBoundBinaryOperatorOperation(BoundBinaryOperator
             Lazy<IOperation> leftOperand = new Lazy<IOperation>(() => Create(boundBinaryOperator.Left));
             Lazy<IOperation> rightOperand = new Lazy<IOperation>(() => Create(boundBinaryOperator.Right));
             IMethodSymbol operatorMethod = boundBinaryOperator.MethodOpt;
+            IMethodSymbol unaryOperatorMethod = null;
+
+            // For dynamic logical operator MethodOpt actually the unary true/false operator
+            if (boundBinaryOperator.Type.IsDynamic() &&
+                (operatorKind == BinaryOperatorKind.ConditionalAnd || operatorKind == BinaryOperatorKind.ConditionalOr) &&
+                operatorMethod?.Parameters.Length == 1)
```

Comment:
Is it possible for there to be default parameters, even in an error scenario?

Reply:
No, signature of operators is fixed.

---
In reply to: [191545921](https://github.com/dotnet/roslyn/pull/27155#discussion_r191545921) [](ancestors = 191545921)